# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1389754569 18000
#      Tue Jan 14 21:56:09 2014 -0500
# Branch stable
# Node ID 8671f39d17286ec20ef8a9c135911e785ea701a8
# Parent  cc1c19863ae3382b90c07f19e913c48156169903
doc: Document missing build tools and external packages (bug #41233)

* install.txi: Document missing build tools gperf and Texinfo and
optional dependency on JDK.

diff --git a/doc/interpreter/install.txi b/doc/interpreter/install.txi
--- a/doc/interpreter/install.txi
+++ b/doc/interpreter/install.txi
@@ -165,16 +165,27 @@ Package for Makefile generation
 (@url{http://www.gnu.org/software/automake}).  Automake is required if
 you modify Octave's @code{Makefile.am} files or other files that they
 depend on.
 
 @item Libtool
 Package for building software libraries
 (@url{http://www.gnu.org/software/libtool}).  Libtool is required by
 Automake.
+
+@item gperf
+Perfect hash function generator (@url{http://www.gnu.org/software/gperf}).
+You will need gperf if you modify the @code{octave.gperf} file or if you
+delete the file that is generated from it.
+
+@item Texinfo
+Package for generating online and print documentation
+(@url{http://www.gnu.org/software/texinfo}).  You will need Texinfo to
+build Octave's documentation or if you modify the documentation source
+files or the docstring of any Octave function.
 @end table
 
 @node External Packages
 @subsection External Packages
 
 The following external packages are required:
 
 @table @asis
@@ -250,16 +261,22 @@ Image processing library (@url{http://ww
 GraphicsMagick++ is used to provide the @code{imread} and @code{imwrite}
 functions.
 
 @item HDF5
 Library for manipulating portable data files
 (@url{http://www.hdfgroup.org/HDF5}).  @sc{hdf5} is required for Octave's
 @code{load} and @code{save} commands to read and write HDF data files.
 
+@item Java Development Kit
+Java programming language compiler and libraries.  The OpenJDK free
+software implementation is recommended (@url{http://openjdk.java.net/}),
+although other JDK implementations may work.  Java is required to be able
+to call Java functions from within Octave.
+
 @item LLVM
 Compiler framework, (@url{http://www.llvm.org}).  LLVM is required for
 Octave's experimental just-in-time (JIT) compilation for speeding up the
 interpreter.
 
 @item OpenGL
 API for portable 2-D and 3-D graphics (@url{http://www.opengl.org}).  An
 OpenGL implementation is required to provide Octave's OpenGL-based
